src/azure-cli/azure/cli/command_modules/storage/_params.py

```diff
@@ -1058,6 +1058,40 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
         c.argument('lease_duration', type=int)
         c.argument('lease_break_period', type=int)
 
+    with self.argument_context('storage container-rm', resource_type=ResourceType.MGMT_STORAGE) as c:
+        from .sdkutil import get_container_access_type_names
+        c.argument('container_name', container_name_type, options_list=('--name', '-n'))
+        c.argument('account_name', help='Storage account name. Related environment variable: AZURE_STORAGE_ACCOUNT.',
+                   completer=get_resource_name_completion_list('Microsoft.Storage/storageAccounts'))
+        c.argument('resource_group_name', required=False, validator=process_resource_group)
+        c.argument('public_access', validator=validate_container_public_access,
+                   arg_type=get_enum_type(get_container_access_type_names()),
+                   help='Specifies whether data in the container may be accessed publicly.')
+        c.ignore('filter', 'maxpagesize')
+
+    with self.argument_context('storage container-rm create', resource_type=ResourceType.MGMT_STORAGE) as c:
+        c.argument('account_name', help='Storage account name. Related environment variable: AZURE_STORAGE_ACCOUNT.')
+        c.argument('fail_on_exist', help='Throw an exception if the container already exists.')
+
+    for item in ['create', 'update']:
+        with self.argument_context('storage container-rm {}'.format(item),
+                                   resource_type=ResourceType.MGMT_STORAGE) as c:
+            c.argument('metadata', nargs='+',
```

Comment:
No need to declare metadata here. See
https://github.com/Azure/azure-cli/blob/2067936fd5c49acce6bd1ed590d8dd25230eb71d/src/azure-cli/azure/cli/command_modules/storage/_params.py#L183-L185

There is inheritance from top level.